src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -121,18 +112,35 @@ export default {
 			openedMenu: this.menuOpen
 		}
 	},
+
 	computed: {
 		collapsible() {
-			return this.item.collapsible && this.item.children && this.item.children.length > 0
+			return !this.forceExpand && this.$slots.children
 		},
 		simpleAction() {
 			if (this.collapsible && !this.item.action) {
 				return this.toggleCollapse
 			} else {
 				return this.item.action
 			}
+		},
+		// Checks if the icon prop is an url
+		isIconUrl() {
+			try {
+				return new URL(this.icon)
+			} catch (error) {
+				return false
+			}
+		},
+		// Checks if the component is already a children of another
+		// instance of AppNavigationItem
+		canHaveChildren() {
+			if (this.$parent.$options._componentTag === 'AppNavigationItem') {
```

Comment:
Does this work if I have a custom component which includes a `AppNavigationItem`?

Reply:
I'm not sure I understand this question, could you give me an example please?

Reply:
Do you mean:

```
<AppNavigationItem />
    <template slot="children">
        <Custom>
            <AppNavigationItem />
        </Custom>
    </template>
<AppNavigationItem />
```

Because if that's the case it wouldn't work :(

Reply:
What I meant was a custom component like this: https://github.com/nextcloud/notes/blob/fc7631eb9e7e89876f2570d18b872b89fb30832c/src/components/NavigationNoteItem.vue

But now, I think there is no problem with the scenario I had in mind.

However, I'm not sure, if this check is really needed. It's there in order to disallow a hierarchy with more than two levels, right?

Reply:
In that case I think it'll work because the root template should not count.
Yes, it prevents `<AppNavigationItem />` children to have their own children by disabling the slot. We could also throw an error if somebody tries to go beyond those 2 levels...

Reply:
> However, I'm not sure, if this check is really needed. It's there in order to disallow a hierarchy with more than two levels, right?

yes :)
We can still adapt this later on if this check doesn't fit your needs ;)